src/AutoMapper/Configuration/MappingExpression.cs

```diff
@@ -239,22 +239,29 @@ private void IgnoreDestinationMember(MemberInfo property)
         public IMappingExpression<TSource, TDestination> Include<TOtherSource, TOtherDestination>()
             where TOtherSource : TSource
             where TOtherDestination : TDestination 
-            => Include(typeof(TOtherSource), typeof(TOtherDestination));
+            => IncludeCore(typeof(TOtherSource), typeof(TOtherDestination));
 
         public IMappingExpression<TSource, TDestination> Include(Type otherSourceType, Type otherDestinationType)
+        {
+            CheckIsDerived(otherSourceType, typeof(TSource));
+            CheckIsDerived(otherDestinationType, typeof(TDestination));
+            return IncludeCore(otherSourceType, otherDestinationType); ;
+        }
+
+        IMappingExpression<TSource, TDestination> IncludeCore(Type otherSourceType, Type otherDestinationType)
         {
-            TypeMapActions.Add(tm => tm.IncludeDerivedTypes(otherSourceType, otherDestinationType));
-
+            TypeMapActions.Add(tm => tm.IncludeDerivedTypes(otherSourceType, otherDestinationType));
             return this;
-        }
+        }
 
         public IMappingExpression<TSource, TDestination> IncludeBase<TSourceBase, TDestinationBase>() 

```

Comment:
We need to constrain TSourceBase, not TSource, and there is no constraint is a base class of. That seems to be the idea.